src/Validator/Rules/PossibleFragmentSpreads.php

```diff
@@ -148,13 +149,12 @@ private function getFragmentType(ValidationContext $context, $name)
         return null;
     }
 
-    public static function typeIncompatibleSpreadMessage($fragName, $parentType, $fragType)
+    public static function typeIncompatibleSpreadMessage($fragName, $parentType, $fragType) : ErrorCode
     {
-        return sprintf(
-            'Fragment "%s" cannot be spread here as objects of type "%s" can never be of type "%s".',
+        return new ErrorCode(ErrorCode::ERR_CANT_SPREAD_FRAGMENT, [
             $fragName,
             $parentType,
-            $fragType
-        );
+            $fragType,
+        ]);
```

Comment:
Nice! One problem is that the error message is not directly reachable from here anymore. So I have to go to `ERR_CANT_SPREAD_FRAGMENT` constant and then literally search for its usage to reach the actual error message.

That's what I meant when mentioned that it will make maintenance and syncing with upstream graphql-js less convenient.

Reply:
> So I have to go to ERR_CANT_SPREAD_FRAGMENT constant and then literally search for its usage to reach the actual error message

I don't understand what you mean. If you're already at this bit of code you can just ctrl+click to jump to the string (well, in my Intellij, anyway) definition. And if you're at the string and want to see where it's used, yes, you can search for it (or in Intelli, "Find Usages"). And having the strings centralized like this also means that we can reuse them (though I haven't checked how much value we would get out of that). If this is going to be a blocker then I can see if I can work out some scheme for each class to manage its own strings, but that's a lot of complexity for the sake of a little convenience and I'm not going to bother unless you're going to insist on it.